From be5b443372359ab598bf0c570959920742948c70 Mon Sep 17 00:00:00 2001
From: Sean Dougherty <sdougherty@microsoft.com>
Date: Wed, 19 Aug 2026 17:11:21 +0000
Subject: [PATCH] container: Add log reader domain

Add an opt-in container_logreader_t domain that permits confined log
collector containers to read host logs without using the broadly
privileged spc_t domain. Keep container_t as the default domain.

Signed-off-by: Sean Dougherty <sdougherty@microsoft.com>
---
 policy/modules/services/container.te | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/policy/modules/services/container.te b/policy/modules/services/container.te
index 97fc614..2fe1b51 100644
--- a/policy/modules/services/container.te
+++ b/policy/modules/services/container.te
@@ -170,6 +170,9 @@ optional_policy(`
 	kubernetes_container(container_t)
 ')
 
+container_domain_template(container_logreader)
+typeattribute container_logreader_t container_system_domain, container_user_domain, container_net_domain;
+
 container_engine_domain_template(container_engine)
 typeattribute container_engine_t container_engine_system_domain;
 type container_engine_exec_t, container_engine_exec_type;
@@ -596,6 +599,21 @@ optional_policy(`
 	rpm_read_db(container_t)
 ')
 
+########################################
+#
+# Log reader container local policy
+#
+
+gen_require(`
+	attribute logfile;
+')
+
+logging_read_all_logs(container_logreader_t)
+logging_read_audit_log(container_logreader_t)
+logging_list_logs(container_logreader_t)
+allow container_logreader_t logfile:lnk_file read_lnk_file_perms;
+allow container_logreader_t container_log_t:file watch;
+
 ########################################
 #
 # Common container engine local policy
